src/bin/pg_autoctl/fsm_transition.c

```diff
@@ -1040,3 +1144,56 @@ fsm_promote_standby(Keeper *keeper)
 
 	return true;
 }
+
+
+/*
+ * fsm_promote_standby_for_primary_maintenance implements the same transition
+ * as the fsm_promote_standby() function, the only difference now is that we
+ * expect the other node to be in the MAINTENANCE state rather than the
+ * DEMOTE_TIMEOUT_STATE.
+ */
+bool
+fsm_promote_standby_for_primary_maintenance(Keeper *keeper)
```

Comment:
minor: Could we change `fsm_promote_standby ` to get the target state and make this one:

```
def fsm_promote_standby_for_primary_maintenance()
    
    fsm_promote_standby(keeper, MAINTENANCE_STATE)
```

Reply:
Nope, or at least, not like that. Look at `fsm.h` where we define that transition function all only take a single argument, that is a pointer to the current Keeper structure.

``` c
/*
 * Each FSM entry is a transition from a current state to the next
 */
typedef bool (*ReachAssignedStateFunction)(Keeper *keeper);

/* defines a possible transition in the FSM */
typedef struct KeeperFSMTransition
{
	NodeState current;
	NodeState assigned;
	const char *comment;
	ReachAssignedStateFunction transitionFunction;
} KeeperFSMTransition;
```

Reply:
If it's really the same except for the other node state. Then we could have something like this:

def fsm_promote_standby_for_primary_maintenance()    
    fsm_promote_standby(keeper, MAINTENANCE_STATE)

def fsm_promote_standby_normal()
    fsm_promote_standby(keeper, DEMOTE_TIMEOUT_STATE)